pkg/operator/csi/csidrivercontrollerservicecontroller/csi_driver_controller_service_controller.go

```diff
@@ -217,6 +231,50 @@ func (c *CSIDriverControllerServiceController) sync(ctx context.Context, syncCon
 	return err
 }
 
+func (c *CSIDriverControllerServiceController) syncDeleting(opSpec *opv1.OperatorSpec, opStatus *opv1.OperatorStatus, ctx context.Context, syncContext factory.SyncContext) error {
+	klog.V(4).Infof("syncDeleting")
+	deployment, err := c.getDeployment(opSpec)
+	if err != nil {
+		return err
+	}
+
+	err = c.kubeClient.AppsV1().Deployments(deployment.Namespace).Delete(ctx, deployment.Namespace, metav1.DeleteOptions{})
+	if err != nil && !apierrors.IsNotFound(err) {
+		return err
+	} else {
+		klog.V(2).Infof("Deleted Deployment %s/%s", deployment.Namespace, deployment.Name)
+	}
+
+	// All removed, remove the finalizer as the last step
+	err = c.operatorClient.RemoveFinalizer(c.name)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *CSIDriverControllerServiceController) getDeployment(opSpec *opv1.OperatorSpec) (*appsv1.Deployment, error) {
+	var clusterID string
+	if c.configInformer != nil {
+		infra, err := c.configInformer.Config().V1().Infrastructures().Lister().Get(infraConfigName)
+		if err != nil {
+			return nil, err
+		}
+		clusterID = infra.Status.InfrastructureName
+	}
+
+	manifest := replacePlaceholders(c.manifest, opSpec, clusterID)
```

Comment:
signature changed to include a clusterid?

Reply:
not in this PR...